articles/how-and-when-to-apply-session-storage-with-JavaScript/index.md

```diff
@@ -0,0 +1,239 @@
+
+#### setItem()
+
+This method is called to store values in the sessionStorage. This method takes in the key and values as parameters.
+```javascript
+sessionStorage.setItem("name:, "John Smith");
+```
+In the above function, `name` is the key, while `John Smith` is the value.
+
+#### getItem()
+
+This function is used to retrieve values from the sessionStorage. It takes in the `key` as a parameter. Using the above example, the key is `name`.
+```javascript
+var user = sessionStorage.getItem("name");
+```
+
+#### removeItem()
+This method also requires a `key` as a parameter. The method will then delete the specified item from the sessionStorage.
+```javascript
+sessionStorage.remove("name");
+```
+When called, the above statement will remove `John Smith` from the sessionStorage.
+
+#### clear()
+
+As the name suggests, this function deletes all items from the sessionStorage. This method does not require any parameters.
+```javascript
+sessionStorage.clear();
+```
+Let&#39;s implement sessionStorage in a real-life application.
+
+### Step 3 – Creating the view
+
+In this step, we will design a simple web page that we will use to interact with the sessionStorage functionalities. The page is designed using HTML 5. The `main.js` file must be imported in the `index.html` file. This allows us to access the required sessionStorage functions. We import the main.js file using the below code snippet.
+
+```html
+  <script type="text/javascript" src="main.js"></script>
+```
+
+It&#39;s important to ensure that all of our buttons have an `id`. This is because we will handle their click events in the `main.js` file.
+
+```html
+<!DOCTYPE html>
+<html>
+
+<head>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width">
+  <title>sessionStorage</title>
+  <script type="text/javascript" src="main.js"></script>
+
+</head>
+
+<body>
+
+  <br>
+  <br>
+  <div class="container" id="formDiv">
+        <form class="form-signin" id="carForm">
+            <h1>Enter details</h1>
+            <label for="carBrand" class="sr-only">Car</label>
+            <input type="text" id="carBrand" class="form-control" placeholder="Car brand" required autofocus><br>
+            <label for="carPrice" class="sr-only">Price</label>
+            <input type="text" id="carPrice" class="form-control" placeholder="Price" required><br>
+            <button class="btn btn-lg btn-primary btn-block" type="submit">Submit</button>
+        </form>
+        <br>
+        <br>
+
+        <button class="btn btn-lg btn-primary btn-block" id="retrieveButton">Retrieve records</button>
+        <br>
+        <div id="retrieve"></div>
+        <br>
+
+        <button class="btn btn-lg btn-danger btn-block" id="removeButton">Remove record</button>
+        <br>
+        <br>
+
+        <button class="btn btn-lg btn-danger btn-block" id="clearButton">Clear all records</button>
+
+  </div>
+
+</body>
+</html>
+
+```
```

Comment:
Hi @lalith1403,
I was reviewing a similar article from @WanjaMIKE and I came here for reference while I noticed this.

A lot of the elements have bootstrap classes which have no impact since there's no bootstrap file linked to the document. I think all the bootstrap class names should be removed to avoid unnecessary confusion for readers.

Reply:
@zolomohan 

Why is the bootstrap making the code snippets wrong? Everything is explained correctly.

Reply:
I'm not saying it's wrong. Since it has no impact on the code, the presence of it is unnecessary in my opinion.

Reply:
@lalith1403 do you agree?

Reply:
I would suggest adding the link tag for bootstrap! The focus is on session storage here. This article explains that in depth. 

A line mentioning that the forms use bootstrap would clear the confusion.

Reply:
That works too!